siva/library.go

```diff
@@ -135,12 +135,21 @@ func (l *Library) Repositories(mode borges.Mode) (borges.RepositoryIterator, err
 
 // Location implements borges.Library interface.
 func (l *Library) Location(id borges.LocationID) (borges.Location, error) {
+	return l.location(id, false)
+}
+
+// AddLocation creates a new borges.Location if it does not exist.
+func (l *Library) AddLocation(id borges.LocationID) (borges.Location, error) {
```

Comment:
Maybe add locations should be used only to create new locations and left `Location` as the way to retrieve an already created location. So if `AddLocation` try to create an already existed siva file then it should return an error. wdyt?